plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -731,7 +731,11 @@ private ConnectorTableMetadata getTableMetadata(ConnectorSession session, Schema
         if (metastore.getTable(new HiveIdentity(session), table.getSchemaName(), table.getTableName()).isEmpty()) {
             throw new TableNotFoundException(table);
         }
+        return getTableMetadataForExistingTable(session, table);
```

Comment:
We can actually just remove the metastore check above, as `getIcebergTable()` already checks the metastore and throws `TableNotFoundException` if not found. No need to introduce a new method.

`getIcebergTable()` calls `loadIcebergTable()` and then does a `operations.current()` on it, which is implemented in `HiveTableOperations` and calls `getTable()` which does the `metastore.getTable()` call (and throws if not found).

Reply:
indeed